sopt/src/test/scala/dagr/sopt/cmdline/ClpReflectiveBuilderTest.scala

```diff
@@ -112,4 +115,28 @@ class ClpReflectiveBuilderTest extends UnitSpec {
       an[IllegalStateException] should be thrownBy _.toCommandLineString
     }
   }
+
+  it should "print v if the argument type was an Option and the value was Some(v)" in {
+    val t = new ClpReflectiveBuilder(classOf[DefaultWithSomeOption])
+    t.argumentLookup.view.foreach {
+      _.toCommandLineString shouldBe "--w 4"
+    }
+  }
+
+  it should "print the special empty or none token for a None value or empty collections" in {
```

Comment:
The idea is to write all the command line options, even those where the user did not override the default values. I prefer printing `:none:` when the collection is empty or the option is `None` because it is explicit.  Otherwise, when looking back on the command line, I don't know if that option defaulted to empty and so was omitted from the printed command line, or if that command line option didn't exist for the given tool/pipeline version.  I'd prefer to make it explicit that the command line option exists and its value is `:none:`.